unittests/runtime-collections/rangemaps.cpp

```diff
@@ -0,0 +1,596 @@
+#include <boost/test/unit_test.hpp>
+
+#include <runtime/collections/rangemap.h>
+
+BOOST_AUTO_TEST_SUITE(RangeMapTest)
+
+BOOST_AUTO_TEST_CASE(rangemap_test_empty) {
+  auto map = RangeMap<int, int>();
+  auto result = map.size();
+  BOOST_CHECK_EQUAL(result, 0);
+  auto result2 = map.containsKey(0);
+  BOOST_CHECK_EQUAL(result2, false);
+}
+
+BOOST_AUTO_TEST_CASE(rangemap_test_concat_success) {
```

Comment:
I don't understand this test. If my understanding is correct, map5 ought to look like:

```
[0, 2) -> 2
[2, 7) -> 1
[8, 9) -> 4
```

So its size ought to be 3, but you assert that it's 1, as if all the ranges got combined. But you can't combine two ranges that map to different values.

Am I missing something?

Reply:
map5 is the concatenation of map1 ( [2,4)->1, [6,7)->1 ) and map4 ( [4,6)->1 ), and therefore is the map [2,7)->1 because all ranges map to the same value. I am reusing map1 here to also test for immutability.